src/libtexture/imagecache.cpp

```diff
@@ -296,14 +297,7 @@ ImageCacheFile::ImageCacheFile(ImageCacheImpl& imagecache,
     // reflected by the fact that m_validspec is false.
 
     // Figure out if it's a UDIM-like virtual texture
-    m_is_udim = (m_filename.find("<UDIM>") != m_filename.npos
-                 || m_filename.find("<U>") != m_filename.npos
-                 || m_filename.find("<V>") != m_filename.npos
-                 || m_filename.find("<u>") != m_filename.npos
-                 || m_filename.find("<v>") != m_filename.npos
-                 || m_filename.find("_u##v##") != m_filename.npos
-                 || m_filename.find("%(UDIM)d") != m_filename.npos)
-                && !Filesystem::exists(m_filename);
+    udim_setup();
```

Comment:
I really like that this now happens early and once only. The run-time spin rw lock always seemed like an unneccessary bottleneck to me.

Reply:
I guess my point is that the number of empty entries should be "somewhat sane" in all cases.
If users put a single UDIM at 1000 and one more at 1, when it's a little bit silly already.... but even in that case, 999 empty udim entries in a vector won't use all their available RAM either.